subsys/net/lib/ocpp/ocpp.c

```diff
@@ -153,11 +153,37 @@ static int ocpp_connect_to_cs(struct ocpp_info *ctx)
 		zsock_close(ui->tcpsock);
 	}
 
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+	ui->tcpsock = zsock_socket(ui->csi.sa_family, SOCK_STREAM, IPPROTO_TLS_1_3);
+	if (ui->tcpsock < 0) {
+		LOG_ERR("Failed to create TLS socket: %d", -errno);
+		return -errno;
+	}
+
+	ret = zsock_setsockopt(ui->tcpsock, SOL_TLS, TLS_SEC_TAG_LIST, ui->csi.creds.sec_tag_list,
+			       ui->csi.creds.sec_tag_list_size);
+	if (ret < 0) {
+		LOG_ERR("Failed to set TLS_SEC_TAG_LIST: %d", -errno);
+		zsock_close(ui->tcpsock);
+		ui->tcpsock = -1;
+		return -errno;
+	}
+
+	ret = zsock_setsockopt(ui->tcpsock, SOL_TLS, TLS_HOSTNAME, ui->csi.creds.tls_hostname,
+			       ui->csi.creds.tls_hostname_size);
+	if (ret < 0) {
+		LOG_ERR("Failed to set TLS_HOSTNAME: %d", -errno);
+		zsock_close(ui->tcpsock);
+		ui->tcpsock = -1;
+		return -errno;
```

Comment:
same here,

We could also share the base string with previous print to save some flash like this
```
LOG_ERR("Failed to set %s: %d", "TLS_HOSTNAME", ret);
```
in which case the previous error print should be changed accordingly.